web-client/src/app/views/triggers/triggers.page.ts

```diff
@@ -70,25 +85,100 @@ export class TriggersPage implements OnInit {
 
   async saveLimits(limitInput: IonInput) {
     const currencyCode = this.selectedOption?.assetDisplay.assetSymbol || '';
-    await this.otpLimitsService.setOtpLimit({
-      currency_code: currencyCode,
-      limit: this.limit,
-    });
 
-    limitInput.value = null;
+    const otpAttempt = await this.otpPromptService.requestOTP();
+    if (!otpAttempt) {
+      return;
+    }
+
+    await withLoadingOverlayOpts(
+      this.loadingCtrl,
+      { message: 'Checking OTP...' },
+      async () => {
+        const otpResult = await this.otpPromptService.checkOtp(otpAttempt);
+        if (otpResult.status === 200) {
+          if (otpResult.data.status === 'approved') {
+            await withLoadingOverlayOpts(
+              this.loadingCtrl,
+              { message: 'Updating OTP Limit...' },
+              async () => {
+                await this.otpLimitsService.setOtpLimit({
+                  currency_code: currencyCode,
+                  limit: this.limit,
+                });
+                this.currentLimitAmount = this.limit;
+                limitInput.value = null;
+              }
+            );
+          } else if (otpResult.data.status === 'pending') {
+            this.notification.showIncorrectOTPWarning();
+          }
+        } else {
+          this.notification.showUnexpectedFailureWarning();
+        }
+      }
+    );
+  }
+
+  /** Validated {@link address}, or `undefined`. */
+  validatedAddress(address: string): string {
+    const trimmed = address?.trim();
+    return trimmed === '' ? '' : trimmed;
```

Comment:
Can you please explain what this does?

Reply:
This function just makes sure the user does not submit an empty wallet address. In the event it does, an 'Invalid Address' popup message appears.

Reply:
But why are you returning `''` ?

Reply:
Because of the `xrpl.isValidAddress()` and `algosdk.isValidAddress()` functions from the respective xrpl and algo SDKs. They only take strings variables when checking the validity of a wallet address. Instead of returning undefined, I am returning an empty (`''`) string, so that I can use the above functions.

Reply:
But `isValidAddress()` will always be `false` if you pass an empty string, why bother passing it?

Reply:
we should check if the address exists, validating an empty string is a pointless exercise, it's like doing `if(false)`

Reply:
Resolved. I've removed the checker that checks for an empty string.